etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/CopyOrcOrderProviderToObrOrderProvider.java

```diff
@@ -0,0 +1,47 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import java.util.Map;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.DiagnosticReport;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.ServiceRequest;
+
+/** Updates the order provider (OBR-16) from the order provider (ORC-12) */
+public class CopyOrcOrderProviderToObrOrderProvider implements CustomFhirTransformation {
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args) {
+        Bundle bundle = (Bundle) resource.getUnderlyingResource();
+        DiagnosticReport diagnosticReport = HapiHelper.getDiagnosticReport(bundle);
+        if (diagnosticReport == null) {
+            return;
+        }
+        ServiceRequest serviceRequest = HapiHelper.getServiceRequest(diagnosticReport);
+        if (serviceRequest == null) {
+            return;
+        }
+        var practitionerRole = HapiHelper.getPractitionerRole(serviceRequest);
+        if (practitionerRole == null) {
+            return;
+        }
+
+        // check if the OBR 16 extension exists
+        var obrExtension = serviceRequest.getExtensionByUrl(HapiHelper.EXTENSION_OBR_URL);
```

Comment:
Could you move the getters and setters for OBR-16 and ORC-12 to the HapiHelper library? You can check what we're doing there for OBR-2, OBR-4, ORC-2, etc as an example. Moving them there will allow us to reuse in other transformations if needed

Reply:
Good call.